lib/tasks/book_image_url.rake

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+desc "Check Book Image URLs"
+task :book_image_url do
+  require "net/http"
+  File.open(File.join(Rails.root, "app", "views", "doc", "_ext_books.erb")).each { |line|
+    # Check every img element to ensuer the src attribute is valid
+    if ((match = line.match(/.*<img\ssrc="([http|https].*)">/)))
+      url = URI.parse(match[1])
+      req = Net::HTTP.new(url.host, url.port)
+      req.use_ssl = match[1].start_with?("https")
+      res = req.request_head(url.path)
+
+      if res.code.to_i == 404
+        puts "[WARN] A book image specified at the following URL was not found: #{match[1]}"
```

Comment:
Sadly this doesn't seem to catch the case fixed recently in #1390. There the server eventually returns an HTTP 200, but it's full of HTML. I wonder if we could check the content-type for something plausible (`image/*`)?

Reply:
Ah, gotcha. I'll add the content-type check to have a little bit more confidence that the return asset is an actual image.